Makefile

```diff
@@ -56,6 +57,17 @@ sync_env: check_poetry_version
 	fi
 	"$(MAKE)" setup_env
 
+.PHONY: fix_omp_issues_for_intel_mac # Fix OMP issues for macOS Intel, https://github.com/zama-ai/concrete-ml-internal/issues/3951
+fix_omp_issues_for_intel_mac:
+	if [[ $$(uname) == "Darwin" ]]; then \
+		WHICH_VENV=$$(which python | sed -e "s@bin/python@@"); \
+		rm $${WHICH_VENV}/lib/python3.9/site-packages/torch/lib/libiomp5.dylib; \
+		ln -s $${WHICH_VENV}/lib/python3.9/site-packages/concrete/.dylibs/libomp.dylib $${WHICH_VENV}/lib/python3.9/site-packages/torch/lib/libiomp5.dylib; \
+		rm $${WHICH_VENV}/lib/python3.9/site-packages/functorch/.dylibs/libiomp5.dylib; \
+		ln -s $${WHICH_VENV}/lib/python3.9/site-packages/concrete/.dylibs/libomp.dylib $${WHICH_VENV}lib/python3.9/site-packages/functorch/.dylibs/libiomp5.dylib; \
```

Comment:
Hum not sure about having python3.9 hard-coded here

Reply:
yes you're right

Reply:
what would you do, instead? I can try to find which python is available, but that's not obvious to me what to do if there is more than a single one. Maybe I can do it for all of them

Reply:
@youben11 : maybe you could guide me to which piece of code, on the Concrete CI, does that? I could see if I could borrow your ideas, if they apply here

Reply:
In our CI, we already setup a specific version of python so we don't really have a problem. For your case, you can run the following command `python -c 'import sys; print(f"{sys.version_info.major}.{sys.version_info.minor}")'`

Reply:
Thx @youben11

Reply:
should be fixed now